corehq/apps/smsforms/models.py

```diff
@@ -208,9 +231,34 @@ def sync_sql_session_from_couch_session(couch_session):
     except SQLXFormsSession.DoesNotExist:
         sql_session = SQLXFormsSession(couch_id=couch_id)
 
-    for attr, value in data.items():
-        setattr(sql_session, attr, value)
+    for attr in SESSION_PROPERTIES_TO_SYNC:
+        setattr(sql_session, attr, data.get(attr, None))
+
+    # hack to avoid excess saves. see sync_couch_session_from_couch_session
+    sql_session.do_not_sync = True
     sql_session.save()
+    sql_session.do_not_sync = False
+
+
+@receiver(post_save, sender=SQLXFormsSession)
+def sync_signal_catcher(sender, instance, *args, **kwargs):
+    sync_couch_session_from_couch_session(instance)
+
+
+def sync_couch_session_from_couch_session(sql_session):
+    if sql_session.do_not_sync:
+        return
+
+    if not sql_session.couch_id:
+        logging.error('Only existing sessions can be synced for now.')
```

Comment:
Could we use notify_exception here instead? Some of this stuff gets called from celery and logging doesn't always cause an email to be sent.